geode-membership/src/main/java/org/apache/geode/distributed/internal/membership/gms/DefaultMembershipStatistics.java

```diff
@@ -17,10 +17,35 @@
 import org.apache.geode.distributed.internal.membership.api.MembershipStatistics;
 
 /**
- * MembeshipStatisticsNoOp is the default implementation of MembershipStatistics. Create
+ * DefaultMembeshipStatistics is the default implementation of MembershipStatistics. Create
  * and install your own if you want to record membership stats.
  */
-public class MembershipStatisticsNoOp implements MembershipStatistics {
+public class DefaultMembershipStatistics implements MembershipStatistics {
+  private long sentBytes;
+  private long ucastWriteBytes;
+  private long ucastReadBytes;
+  private long mcastWriteBytes;
+  private long mcastReadBytes;
+  private long ucastRetransmits;
+  private long mcastRetransmits;
+  private long mcastRetransmitRequests;
+  private long heartbeatRequestsSent;
+  private long heartbeatRequestsReceived;
+  private long heartbeatsSent;
+  private long heartbeatsReceived;
+  private long suspectsSent;
+  private long suspectsReceived;
+  private long finalCheckRequestsSent;
+  private long finalCheckRequestsReceived;
+  private long finalCheckResponsesSent;
+  private long finalCheckResponsesReceived;
+  private long tcpFinalCheckRequestsSent;
+  private long tcpFinalCheckRequestsReceived;
+  private long tcpFinalCheckResponsesSent;
+  private long tcpFinalCheckResponsesReceived;
+  private long udpFinalCheckRequestsSent;
+  private long udpFinalCheckResponsesReceived;
```

Comment:
Where are these stats being used? Is it ok that they are not thread-safe?

Reply:
@Bill The stats are being used in unit tests

Reply:
Am I wrong to be concerned about synchronization on these fields?